contracts/lockdrop/src/contract.rs

```diff
@@ -392,8 +455,427 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    let new_pool_info: PoolInfoV2 = ASSET_POOLS_V2.load(deps.storage, pool_type)?;
+    let new_pool_addr = get_lp_token_pool_addr(deps.as_ref(), &new_pool_info.lp_token)?;
+    let mut msgs = vec![];
+    let current_ntrn_balance = deps
+        .querier
+        .query_balance(&env.contract.address, UNTRN_DENOM)?
+        .amount;
+    attrs.push(attr(
+        "ntrn_balance_change",
+        (current_ntrn_balance - prev_ntrn_balance).to_string(),
+    ));
+
+    pool_info.generator_ntrn_per_share = pool_info.generator_ntrn_per_share.checked_add({
+        let reward_token_balance = get_reward_amount(
+            deps.as_ref(),
+            &env,
+            &pool_info.lp_token.to_string(),
+            &generator.to_string(),
+        )?;
+        attrs.push(attr(
+            "reward_token_balance",
+            reward_token_balance.to_string(),
+        ));
+        let base_reward_received = reward_token_balance.checked_sub(prev_reward_amount)?;
+        Decimal::from_ratio(base_reward_received, pool_info.amount_in_lockups)
+    })?;
+    attrs.push(attr(
+        "generator_ntrn_per_share",
+        pool_info.generator_ntrn_per_share.to_string(),
+    ));
+    ASSET_POOLS.save(deps.storage, pool_type, &pool_info, env.block.height)?;
+
+    let ntrn_to_new_pool = current_ntrn_balance - prev_ntrn_balance;
+    let new_pool_info: astroport::pair::PoolResponse = deps
+        .querier
+        .query_wasm_smart(&new_pool_addr, &astroport::pair::QueryMsg::Pool {})?;
+    let token_denom = new_pool_info
+        .assets
+        .iter()
+        .find_map(|x| match &x.info {
+            AssetInfo::NativeToken { denom } if denom != UNTRN_DENOM => Some(denom.clone()),
+            _ => None,
+        })
+        .ok_or_else(|| StdError::generic_err("No second leg of pair found"))?;
+    attrs.push(attr("token_denom", token_denom.clone()));
+    let token_balance = deps
+        .querier
+        .query_balance(&env.contract.address, token_denom.as_str())?
+        .amount;
+    attrs.push(attr("token_balance", token_balance.to_string()));
+
+    let base = Asset {
+        amount: ntrn_to_new_pool,
+        info: AssetInfo::NativeToken {
+            denom: UNTRN_DENOM.to_string(),
+        },
+    };
+    let other = Asset {
+        amount: token_balance,
+        info: AssetInfo::NativeToken {
+            denom: token_denom.to_string(),
+        },
+    };
+    let mut funds = vec![
+        Coin {
+            denom: UNTRN_DENOM.to_string(),
+            amount: ntrn_to_new_pool,
+        },
+        Coin {
+            denom: token_denom,
+            amount: token_balance,
+        },
+    ];
+
+    funds.sort_by(|a, b| a.denom.cmp(&b.denom));
+    msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: new_pool_addr,
+        funds,
+        msg: to_binary(&astroport::pair::ExecuteMsg::ProvideLiquidity {
+            assets: vec![base, other],
+            slippage_tolerance,
+            auto_stake: None,
```

Comment:
btw don't we want to autostake? I set it to None, too, but now I wonder why

Reply:
I see, you stake at the next step. but I don't :) need to fix it I think

Reply:
wanted to be more explicit and keep the way it's used in auction/lockdrop. But if you insist will change. wdyt @pr0n00gler @swelf19